src/main/java/com/redhat/cloud/common/clowder/configsource/ClowderConfigSource.java

```diff
@@ -404,6 +405,86 @@ public String getValue(String configKey) {
                     throw e;
                 }
             }
+
+            if (configKey.startsWith(CLOWDER_PRIVATE_ENDPOINTS)) {
```

Comment:
Would it be possible to have a function that deals with the common logic? It's likely that this will be updated (say, to add a new parameter that will be common to both private and public endpoints.

Reply:
Yes, I was doubting whether to extract the common logic or not because of the parameter difference that we saw earlier, but let me refactor it real quick

Reply:
I found a way by making the `PrivateEndpointConfig` extend the `EndpointConfig`, but I am not sure if this is as clean as you'd expect?